### PR TITLE
fix: improve getPathName

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

file_loader, load_controller

##### Description of change

pathname can be got from properties, so don't parse from fullpath again

Closes https://github.com/eggjs/egg/issues/472

### CODE REVIEW

**Comment by popomore** [comment]
换了一种实现

File: lib/loader/file_loader.js [hunk 1]
```diff
@@ -132,15 +132,17 @@ class FileLoader {
...
         const fullpath = path.join(directory, filepath);
-        if (!fs.statSync(fullpath).isFile()) {
-          continue;
-        }
+        if (!fs.statSync(fullpath).isFile()) continue;
+        // get properties
+        // app/service/foo/bar.js => [ 'foo', 'bar' ]
         const properties = getProperties(filepath, this.options);
-        const exports = getExports(fullpath, this.options.initializer, this.options.call, this.options.inject);
+        // app/service/foo/bar.js => service.foo.bar
```
Comment (fengmk2) [inline lib/loader/file_loader.js]: windows 也是 `/` 分隔？还是说 globby 做了统一处理？

Comment (popomore) [inline lib/loader/file_loader.js]: https://github.com/isaacs/node-glob#windows

应该是 glob 处理的

**Comment by popomore** [comment]
测试过了

**Comment by dead-horse** [comment]
和之前的一起发了 2.2.0

**Pull request merged**